apps/web-giddh/src/app/accounting/journal-voucher/journal-voucher.component.ts

```diff
@@ -0,0 +1,277 @@
+    //     keyCode: 118, // 'F7',
+    //     key: FUNCTIONAL_KEYS.F7,
+    //     inputForFn: {
+    //         page: 'Journal',
+    //         uniqueName: 'purchases',
+    //         gridType: 'voucher'
+    //     }
+    // }, {
+    //     keyCode: 119, // 'F8',
+    //     key: FUNCTIONAL_KEYS.F8,
+    //     inputForFn: {
+    //         page: 'Sales',
+    //         uniqueName: 'purchases',
+    //         gridType: 'voucher'
+    //     }
+    // }, {
+    //     keyCode: 119, // 'F8',
+    //     key: FUNCTIONAL_KEYS.F8,
+    //     altKey: true,
+    //     inputForFn: {
+    //         page: 'Credit note',
+    //         uniqueName: 'purchases',
+    //         gridType: 'voucher'
+    //     }
+    // }, {
+    //     keyCode: 120, // 'F9',
+    //     key: FUNCTIONAL_KEYS.F9,
+    //     inputForFn: {
+    //         page: 'Purchase',
+    //         uniqueName: 'purchases',
+    //         gridType: 'voucher'
+    //     }
+    // }, {
+    //     keyCode: 120, // 'F9',
+    //     key: FUNCTIONAL_KEYS.F9,
+    //     altKey: true,
+    //     inputForFn: {
+    //         page: 'Debit note',
+    //         uniqueName: 'purchases',
+    //         gridType: 'voucher'
+    //     }
+    // }
+];
+
+/** Pages which have Voucher and Invoice as options */
+export const PAGES_WITH_CHILD = ['Purchase', 'Sales', 'Credit note', 'Debit note'];
+
+@Component({
+    templateUrl: './journal-voucher.component.html',
+    styleUrls: ['./journal-voucher.component.scss']
+})
+export class JournalVoucherComponent implements OnInit, OnDestroy {
+
+    public gridType: string = 'voucher';
+    public selectedPage: string = 'Contra';
+    public flattenAccounts: any = [];
+    public openDatePicker: boolean = false;
+    public openCreateAccountPopupInVoucher: boolean = false;
+    public openCreateAccountPopupInInvoice: boolean = false;
+    public saveEntryInVoucher: boolean = false;
+    public saveEntryInInvoice: boolean = false;
+    /** Current date to show the balance till date */
+    public currentDate: string;
+
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+
+    /** @ignore */
+    constructor(
+        private store: Store<AppState>,
+        private companyActions: CompanyActions,
+        private tallyModuleService: TallyModuleService,
+        private accountService: AccountService,
+        private sidebarAction: SidebarAction
+    ) {
+        this.tallyModuleService.selectedPageInfo.subscribe((data) => {
+            if (data) {
+                this.gridType = data.gridType;
+                this.selectedPage = data.page;
+            }
+        });
+    }
+
+    @HostListener('document:keydown', ['$event'])
+    public beforeunloadHandler(event: KeyboardEvent) {
+        return event.key !== FUNCTIONAL_KEYS.F5;
+    }
+
+    @HostListener('document:keyup', ['$event'])
+    public handleKeyboardEvent(event: KeyboardEvent) {
+        if (event.ctrlKey && event.key.toLowerCase() === 'a') { // Ctrl + A
```

Comment:
nullity check for event

Reply:
@rohit-walkover Not required as host listener is used and no custom calls are made